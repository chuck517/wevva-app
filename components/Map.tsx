import React from 'react';
import { View, Text } from 'react-native';
import MapView, { UrlTile } from 'react-native-maps';
import { styles } from '../styles/styles';
import nightMapStyle from '../styles/nightMapStyle.json';
const EXPO_API_KEY_OWM = '57311a90a3e7cbf52c5f885c10d6c755';


// Displays map of location with Open Weather Map tiles on top showing areas of rain. Quality is generally bad however
// this is the only free one I could find available as of today.

export default function Map({ lat, lon, timeOfDay }) {
  return (
    <>
      <Text style={styles.titleText}>Precipitation Map</Text>
      <View style={styles.mapContainer}>
        <MapView
          customMapStyle={timeOfDay == 'd' ? [] : nightMapStyle}
          style={styles.map}
          initialRegion={{
            latitude: lat,
            longitude: lon,
            latitudeDelta: 3,
            longitudeDelta: 3,
          }}
        >
          <UrlTile
            key="weatherTile"
            urlTemplate={`https://tile.openweathermap.org/map/precipitation_new/{z}/{x}/{y}.png?appid=${EXPO_API_KEY_OWM}`}
          />
        </MapView>
      </View>
    </>
  );
}
