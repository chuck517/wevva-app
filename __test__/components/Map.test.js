import React from 'react';
import renderer from 'react-test-renderer';
import Map from '../../components/Map';


it('renders without crashing', () => {
  const rendered = renderer.create(<Map />).toJSON();
  expect(rendered).toBeTruthy();
});

it('test against snapshot', () => {
  const tree = renderer.create(<Map />).toJSON();
  expect(tree).toMatchSnapshot();
});