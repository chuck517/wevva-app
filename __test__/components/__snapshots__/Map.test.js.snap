// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> renders correctly across screens 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 17,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Precipitation Map
  </Text>,
  <View
    style={
      Object {
        "borderRadius": 7,
        "height": 300,
        "marginBottom": 30,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <AIRMap
      customMapStyle={
        Array [
          Object {
            "elementType": "geometry",
            "stylers": Array [
              Object {
                "color": "#242f3e",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "stylers": Array [
              Object {
                "color": "#746855",
              },
            ],
          },
          Object {
            "elementType": "labels.text.stroke",
            "stylers": Array [
              Object {
                "color": "#242f3e",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "administrative.locality",
            "stylers": Array [
              Object {
                "color": "#d59563",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "poi",
            "stylers": Array [
              Object {
                "color": "#d59563",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "poi.park",
            "stylers": Array [
              Object {
                "color": "#263c3f",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "poi.park",
            "stylers": Array [
              Object {
                "color": "#6b9a76",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "road",
            "stylers": Array [
              Object {
                "color": "#38414e",
              },
            ],
          },
          Object {
            "elementType": "geometry.stroke",
            "featureType": "road",
            "stylers": Array [
              Object {
                "color": "#212a37",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "road",
            "stylers": Array [
              Object {
                "color": "#9ca5b3",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "road.highway",
            "stylers": Array [
              Object {
                "color": "#746855",
              },
            ],
          },
          Object {
            "elementType": "geometry.stroke",
            "featureType": "road.highway",
            "stylers": Array [
              Object {
                "color": "#1f2835",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "road.highway",
            "stylers": Array [
              Object {
                "color": "#f3d19c",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "transit",
            "stylers": Array [
              Object {
                "color": "#2f3948",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "transit.station",
            "stylers": Array [
              Object {
                "color": "#d59563",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "water",
            "stylers": Array [
              Object {
                "color": "#17263c",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "water",
            "stylers": Array [
              Object {
                "color": "#515c6d",
              },
            ],
          },
          Object {
            "elementType": "labels.text.stroke",
            "featureType": "water",
            "stylers": Array [
              Object {
                "color": "#17263c",
              },
            ],
          },
        ]
      }
      handlePanDrag={false}
      initialRegion={
        Object {
          "latitude": undefined,
          "latitudeDelta": 3,
          "longitude": undefined,
          "longitudeDelta": 3,
        }
      }
      onChange={[Function]}
      onLayout={[Function]}
      onMapReady={[Function]}
      onMarkerPress={[Function]}
      region={null}
      style={
        Object {
          "height": 300,
          "width": "100%",
        }
      }
    >
      <AIRMapUrlTile
        urlTemplate="https://tile.openweathermap.org/map/precipitation_new/{z}/{x}/{y}.png?appid=5d6cc031484aebd5fabaa15febdd9236"
      />
    </AIRMap>
  </View>,
]
`;

exports[`test against snapshot 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 17,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Precipitation Map
  </Text>,
  <View
    style={
      Object {
        "borderRadius": 7,
        "height": 300,
        "marginBottom": 30,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <AIRMap
      customMapStyle={
        Array [
          Object {
            "elementType": "geometry",
            "stylers": Array [
              Object {
                "color": "#242f3e",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "stylers": Array [
              Object {
                "color": "#746855",
              },
            ],
          },
          Object {
            "elementType": "labels.text.stroke",
            "stylers": Array [
              Object {
                "color": "#242f3e",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "administrative.locality",
            "stylers": Array [
              Object {
                "color": "#d59563",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "poi",
            "stylers": Array [
              Object {
                "color": "#d59563",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "poi.park",
            "stylers": Array [
              Object {
                "color": "#263c3f",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "poi.park",
            "stylers": Array [
              Object {
                "color": "#6b9a76",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "road",
            "stylers": Array [
              Object {
                "color": "#38414e",
              },
            ],
          },
          Object {
            "elementType": "geometry.stroke",
            "featureType": "road",
            "stylers": Array [
              Object {
                "color": "#212a37",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "road",
            "stylers": Array [
              Object {
                "color": "#9ca5b3",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "road.highway",
            "stylers": Array [
              Object {
                "color": "#746855",
              },
            ],
          },
          Object {
            "elementType": "geometry.stroke",
            "featureType": "road.highway",
            "stylers": Array [
              Object {
                "color": "#1f2835",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "road.highway",
            "stylers": Array [
              Object {
                "color": "#f3d19c",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "transit",
            "stylers": Array [
              Object {
                "color": "#2f3948",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "transit.station",
            "stylers": Array [
              Object {
                "color": "#d59563",
              },
            ],
          },
          Object {
            "elementType": "geometry",
            "featureType": "water",
            "stylers": Array [
              Object {
                "color": "#17263c",
              },
            ],
          },
          Object {
            "elementType": "labels.text.fill",
            "featureType": "water",
            "stylers": Array [
              Object {
                "color": "#515c6d",
              },
            ],
          },
          Object {
            "elementType": "labels.text.stroke",
            "featureType": "water",
            "stylers": Array [
              Object {
                "color": "#17263c",
              },
            ],
          },
        ]
      }
      handlePanDrag={false}
      initialRegion={
        Object {
          "latitude": undefined,
          "latitudeDelta": 3,
          "longitude": undefined,
          "longitudeDelta": 3,
        }
      }
      onChange={[Function]}
      onLayout={[Function]}
      onMapReady={[Function]}
      onMarkerPress={[Function]}
      region={null}
      style={
        Object {
          "height": 300,
          "width": "100%",
        }
      }
    >
      <AIRMapUrlTile
        urlTemplate="https://tile.openweathermap.org/map/precipitation_new/{z}/{x}/{y}.png?appid=5d6cc031484aebd5fabaa15febdd9236"
      />
    </AIRMap>
  </View>,
]
`;
